Build fails in "strictTemplates" mode for <igx-column [dataType]="'number'">

## Description
Setting this in `tsconfig.json`
```json
{
  "angularCompilerOptions": {
    "enableIvy": true,
    "fullTemplateTypeCheck": true,
    "preserveWhitespaces": false,
    "strictInjectionParameters": true,
    "strictTemplates": true
  },
 .......
}
```
causes a build error for
```html
<igx-column [header]="'Total'" [field]="'total'" [filterable]="true" [dataType]="'number'"> </igx-column>
```
Setting `"strictTemplates": false` avoids the build failure described above.

 * igniteui-angular version: 9.0.0-rc.2
 * browser: build issue

## Steps to reproduce  

1. Try with `"strictTemplates": true`
2. Experience the error
3. Try with `"strictTemplates": false`
4. The build does not stumble.

## Result
Build failure
```log
src/app/pages/home/statistics/statistics-monthly.component.html:13:88 - error TS2322: Type '"number"' is not assignable to type 'DataType'.

13       <igx-column [header]="'Total'" [field]="'total'" [filterable]="true" [dataType]="'number'"> </igx-column>
```

## Expected result
Strict template mode should be possible.

--- a/projects/igniteui-angular/src/lib/avatar/avatar.component.ts
+++ b/projects/igniteui-angular/src/lib/avatar/avatar.component.ts
@@ -10,21 +10,24 @@ import {
     ViewChild
 } from '@angular/core';
 import { IgxIconModule } from '../icon/public_api';
+import { mkenum } from '../core/utils';
 
 let NEXT_ID = 0;
+export const IgxAvatarSize = mkenum({
+    SMALL: 'small',
+    MEDIUM: 'medium',
+    LARGE: 'large'
+});
+export type IgxAvatarSize = (typeof IgxAvatarSize)[keyof typeof IgxAvatarSize];
 
-export enum IgxAvatarSize {
-    SMALL = 'small',
-    MEDIUM = 'medium',
-    LARGE = 'large'
-}
+export const IgxAvatarType = mkenum({
+    INITIALS: 'initials',
+    IMAGE: 'image',
+    ICON: 'icon',
+    CUSTOM: 'custom'
+});
 
-export enum IgxAvatarType {
-    INITIALS = 'initials',
-    IMAGE = 'image',
-    ICON = 'icon',
-    CUSTOM = 'custom',
-}
+export type IgxAvatarType = (typeof IgxAvatarType)[keyof typeof IgxAvatarType];
 
 /**
  * Avatar provides a way to display an image, icon or initials to the user.

--- a/projects/igniteui-angular/src/lib/badge/badge.component.ts
+++ b/projects/igniteui-angular/src/lib/badge/badge.component.ts
@@ -1,19 +1,21 @@
 import { CommonModule } from '@angular/common';
 import { Component, HostBinding, Input, NgModule } from '@angular/core';
 import { IgxIconModule } from '../icon/public_api';
+import { mkenum } from '../core/utils';
 
 let NEXT_ID = 0;
 
 /**
  * Determines the igxBadge type
  */
-export enum IgxBadgeType {
-    PRIMARY = 'primary',
-    INFO = 'info',
-    SUCCESS = 'success',
-    WARNING = 'warning',
-    ERROR = 'error'
-}
+export const IgxBadgeType = mkenum({
+    PRIMARY: 'primary',
+    INFO: 'info',
+    SUCCESS: 'success',
+    WARNING: 'warning',
+    ERROR: 'error'
+});
+export type IgxBadgeType = (typeof IgxBadgeType)[keyof typeof IgxBadgeType];
 /**
  * Badge provides visual notifications used to decorate avatars, menus, etc.
  *

--- a/projects/igniteui-angular/src/lib/calendar/calendar-base.ts
+++ b/projects/igniteui-angular/src/lib/calendar/calendar-base.ts
@@ -3,16 +3,17 @@ import { WEEKDAYS, Calendar, isDateInRanges, IFormattingOptions, IFormattingView
 import { ControlValueAccessor } from '@angular/forms';
 import { DateRangeDescriptor } from '../core/dates';
 import { Subject } from 'rxjs';
-import { isDate } from '../core/utils';
+import { isDate, mkenum } from '../core/utils';
 
 /**
- * Sets the selction type - single, multi or range.
+ * Sets the selection type - single, multi or range.
  */
-export enum CalendarSelection {
-    SINGLE = 'single',
-    MULTI = 'multi',
-    RANGE = 'range'
-}
+export const CalendarSelection = mkenum({
+    SINGLE: 'single',
+    MULTI: 'multi',
+    RANGE: 'range'
+});
+export type CalendarSelection = (typeof CalendarSelection)[keyof typeof CalendarSelection];
 
 export enum ScrollMonth {
     PREV = 'prev',

--- a/projects/igniteui-angular/src/lib/card/card.component.ts
+++ b/projects/igniteui-angular/src/lib/card/card.component.ts
@@ -12,6 +12,7 @@ import {
     SimpleChanges
 } from '@angular/core';
 import { IgxButtonModule } from '../directives/button/button.directive';
+import { mkenum } from '../core/utils';
 
 let NEXT_ID = 0;
 
@@ -200,10 +201,11 @@ export class IgxCardFooterDirective {
  * ```
  */
 
-export enum IgxCardType {
-    ELEVATED = 'elevated',
-    OUTLINED = 'outlined'
-}
+export const IgxCardType = mkenum({
+    ELEVATED: 'elevated',
+    OUTLINED: 'outlined'
+});
+export type IgxCardType = (typeof IgxCardType)[keyof typeof IgxCardType];
 
 @Component({
     selector: 'igx-card',
@@ -276,10 +278,11 @@ export class IgxCardComponent {
     public horizontal = false;
 }
 
-export enum IgxCardActionsLayout {
-    START = 'start',
-    JUSTIFY = 'justify',
-}
+export const IgxCardActionsLayout = mkenum({
+    START: 'start',
+    JUSTIFY: 'justify'
+});
+export type IgxCardActionsLayout = (typeof IgxCardActionsLayout)[keyof typeof IgxCardActionsLayout];
 
 /**
  * IgxCardActions is container for the card actions.

--- a/projects/igniteui-angular/src/lib/carousel/carousel.component.ts
+++ b/projects/igniteui-angular/src/lib/carousel/carousel.component.ts
@@ -21,7 +21,7 @@ import {
     Injectable
 } from '@angular/core';
 import { IgxIconModule } from '../icon/public_api';
-import { IBaseEventArgs, PlatformUtil } from '../core/utils';
+import { IBaseEventArgs, mkenum, PlatformUtil } from '../core/utils';
 import { Subject, merge } from 'rxjs';
 import { takeUntil } from 'rxjs/operators';
 import { IgxCarouselIndicatorDirective, IgxCarouselNextButtonDirective, IgxCarouselPrevButtonDirective } from './carousel.directives';
@@ -34,16 +34,18 @@ import { HammerGestureConfig, HAMMER_GESTURE_CONFIG } from '@angular/platform-br
 
 let NEXT_ID = 0;
 
-export enum CarouselIndicatorsOrientation {
-    bottom = 'bottom',
-    top = 'top'
-}
+export const CarouselIndicatorsOrientation = mkenum({
+    bottom: 'bottom',
+    top: 'top'
+});
+export type CarouselIndicatorsOrientation = (typeof CarouselIndicatorsOrientation)[keyof typeof CarouselIndicatorsOrientation];
 
-export enum CarouselAnimationType {
-    none = 'none',
-    slide = 'slide',
-    fade = 'fade'
-}
+export const CarouselAnimationType = mkenum({
+    none: 'none',
+    slide: 'slide',
+    fade: 'fade'
+});
+export type CarouselAnimationType = (typeof CarouselAnimationType)[keyof typeof CarouselAnimationType];
 
 export interface CarouselAnimationSettings {
     enterAnimation: AnimationReferenceMetadata;

--- a/projects/igniteui-angular/src/lib/checkbox/checkbox.component.ts
+++ b/projects/igniteui-angular/src/lib/checkbox/checkbox.component.ts
@@ -13,13 +13,14 @@ import {
 } from '@angular/core';
 import { CheckboxRequiredValidator, ControlValueAccessor, NG_VALIDATORS, NG_VALUE_ACCESSOR } from '@angular/forms';
 import { IgxRippleModule } from '../directives/ripple/ripple.directive';
-import { isIE, IBaseEventArgs } from '../core/utils';
+import { isIE, IBaseEventArgs, mkenum } from '../core/utils';
 import { EditorProvider } from '../core/edit-provider';
 
-export enum LabelPosition {
-    BEFORE = 'before',
-    AFTER = 'after'
-}
+export const LabelPosition = mkenum({
+    BEFORE: 'before',
+    AFTER: 'after'
+});
+export type LabelPosition = (typeof LabelPosition)[keyof typeof LabelPosition];
 
 export interface IChangeCheckboxEventArgs extends IBaseEventArgs {
     checked: boolean;

--- a/projects/igniteui-angular/src/lib/core/displayDensity.ts
+++ b/projects/igniteui-angular/src/lib/core/displayDensity.ts
@@ -1,15 +1,15 @@
 import { InjectionToken, Input, Output, EventEmitter, DoCheck, OnInit, Directive, NgModule, Optional, Inject } from '@angular/core';
-import { IBaseEventArgs } from './utils';
-
+import { IBaseEventArgs, mkenum } from './utils';
 
 /**
- * Defines the posible values of the components' display density.
+ * Defines the possible values of the components' display density.
  */
-export enum DisplayDensity {
-    comfortable = 'comfortable',
-    cosy = 'cosy',
-    compact = 'compact'
-}
+export const DisplayDensity = mkenum({
+    comfortable: 'comfortable',
+    cosy: 'cosy',
+    compact: 'compact'
+});
+export type DisplayDensity = (typeof DisplayDensity)[keyof typeof DisplayDensity];
 
 /**
  * Describes the object used to configure the DisplayDensity in Angular DI.

--- a/projects/igniteui-angular/src/lib/core/utils.ts
+++ b/projects/igniteui-angular/src/lib/core/utils.ts
@@ -400,3 +400,5 @@ export function yieldingLoop(count: number, chunkSize: number, callback: (index:
     };
     chunk();
 }
+
+export function mkenum<T extends { [index: string]: U }, U extends string>(x: T) { return x; }

--- a/projects/igniteui-angular/src/lib/data-operations/data-util.ts
+++ b/projects/igniteui-angular/src/lib/data-operations/data-util.ts
@@ -12,19 +12,20 @@ import { IGroupingState } from './groupby-state.interface';
 import { ISortingExpression } from './sorting-expression.interface';
 import { FilteringStrategy } from './filtering-strategy';
 import { ITreeGridRecord } from '../grids/tree-grid/public_api';
-import { cloneValue, mergeObjects } from '../core/utils';
+import { cloneValue, mergeObjects, mkenum } from '../core/utils';
 import { Transaction, TransactionType, HierarchicalTransaction } from '../services/transaction/transaction';
 import { getHierarchy, isHierarchyMatch } from './operations';
 
 /**
  * @hidden
  */
-export enum DataType {
-    String = 'string',
-    Number = 'number',
-    Boolean = 'boolean',
-    Date = 'date'
-}
+export const DataType = mkenum({
+    String: 'string',
+    Number: 'number',
+    Boolean: 'boolean',
+    Date: 'date'
+});
+export type DataType = (typeof DataType)[keyof typeof DataType];
 
 /**
  * @hidden

--- a/projects/igniteui-angular/src/lib/date-picker/date-picker.component.ts
+++ b/projects/igniteui-angular/src/lib/date-picker/date-picker.component.ts
@@ -54,7 +54,7 @@ import {
 } from './date-picker.utils';
 import { DatePickerDisplayValuePipe, DatePickerInputValuePipe } from './date-picker.pipes';
 import { IDatePicker } from './date-picker.common';
-import { KEYS, CancelableBrowserEventArgs, isIE, isEqual, IBaseEventArgs } from '../core/utils';
+import { KEYS, CancelableBrowserEventArgs, isIE, isEqual, IBaseEventArgs, mkenum } from '../core/utils';
 import { IgxDatePickerTemplateDirective, IgxDatePickerActionsDirective } from './date-picker.directives';
 import { IgxCalendarContainerComponent } from './calendar-container.component';
 import { InteractionMode } from '../core/enums';
@@ -107,12 +107,13 @@ export interface IFormatOptions {
  * 'longDate': equivalent to 'MMMM d, y' (June 15, 2015).
  * 'fullDate': equivalent to 'EEEE, MMMM d, y' (Monday, June 15, 2015).
  */
-export enum PredefinedFormatOptions {
-    ShortDate = 'shortDate',
-    MediumDate = 'mediumDate',
-    LongDate = 'longDate',
-    FullDate = 'fullDate'
-}
+export const PredefinedFormatOptions = mkenum({
+    ShortDate: 'shortDate',
+    MediumDate: 'mediumDate',
+    LongDate: 'longDate',
+    FullDate: 'fullDate'
+});
+export type PredefinedFormatOptions = (typeof PredefinedFormatOptions)[keyof typeof PredefinedFormatOptions];
 
 const noop = () => { };
 

--- a/projects/igniteui-angular/src/lib/directives/divider/divider.directive.ts
+++ b/projects/igniteui-angular/src/lib/directives/divider/divider.directive.ts
@@ -1,9 +1,11 @@
 import { Directive, HostBinding, NgModule, Input } from '@angular/core';
+import { mkenum } from '../../core/utils';
 
-export enum IgxDividerType {
-    SOLID = 'solid',
-    DASHED = 'dashed'
-}
+export const IgxDividerType = mkenum({
+    SOLID: 'solid',
+    DASHED: 'dashed'
+});
+export type IgxDividerType = (typeof IgxDividerType)[keyof typeof IgxDividerType];
 
 let NEXT_ID = 0;
 

--- a/projects/igniteui-angular/src/lib/drop-down/drop-down.common.ts
+++ b/projects/igniteui-angular/src/lib/drop-down/drop-down.common.ts
@@ -1,4 +1,4 @@
-import { CancelableEventArgs, CancelableBrowserEventArgs, IBaseEventArgs } from '../core/utils';
+import { CancelableEventArgs, CancelableBrowserEventArgs, IBaseEventArgs, mkenum } from '../core/utils';
 import { IgxDropDownItemBaseDirective } from './drop-down-item.base';
 import { IToggleView } from '../core/navigation/IToggleView';
 import { EventEmitter, InjectionToken } from '@angular/core';
@@ -11,11 +11,12 @@ export enum Navigate {
 }
 
 /** Key actions that have designated handlers in IgxDropDownComponent */
-export enum DropDownActionKey {
-    ESCAPE = 'escape',
-    ENTER = 'enter',
-    SPACE = 'space'
-}
+export const DropDownActionKey = mkenum({
+    ESCAPE: 'escape',
+    ENTER: 'enter',
+    SPACE: 'space'
+});
+export type DropDownActionKey = (typeof DropDownActionKey)[keyof typeof DropDownActionKey];
 
 /**
  * Interface that encapsulates onSelection event arguments - old selection, new selection and cancel selection.

--- a/projects/igniteui-angular/src/lib/expansion-panel/expansion-panel-header.component.ts
+++ b/projects/igniteui-angular/src/lib/expansion-panel/expansion-panel-header.component.ts
@@ -13,15 +13,17 @@ import {
 } from '@angular/core';
 import { IgxExpansionPanelIconDirective } from './expansion-panel.directives';
 import { IExpansionPanelEventArgs, IGX_EXPANSION_PANEL_COMPONENT, IgxExpansionPanelBase } from './expansion-panel.common';
+import { mkenum } from '../core/utils';
 
 /**
  * @hidden
  */
-export enum ICON_POSITION {
-    LEFT = 'left',
-    NONE = 'none',
-    RIGHT = 'right'
-}
+export const ICON_POSITION = mkenum({
+    LEFT: 'left',
+    NONE: 'none',
+    RIGHT: 'right'
+});
+export type ICON_POSITION = (typeof ICON_POSITION)[keyof typeof ICON_POSITION];
 
 
 @Component({

--- a/projects/igniteui-angular/src/lib/grids/column-chooser-base.ts
+++ b/projects/igniteui-angular/src/lib/grids/column-chooser-base.ts
@@ -205,7 +205,7 @@ export abstract class ColumnChooserBaseDirective implements OnDestroy {
     /**
      * @hidden
      */
-    private _columnDisplayOrder = ColumnDisplayOrder.DisplayOrder;
+    private _columnDisplayOrder: ColumnDisplayOrder = ColumnDisplayOrder.DisplayOrder;
     /**
      * @hidden
      */

--- a/projects/igniteui-angular/src/lib/grids/common/enums.ts
+++ b/projects/igniteui-angular/src/lib/grids/common/enums.ts
@@ -1,19 +1,23 @@
 
-export enum FilterMode {
-    quickFilter = 'quickFilter',
-    excelStyleFilter = 'excelStyleFilter'
-}
+import { mkenum } from '../../core/utils';
+export const FilterMode = mkenum({
+    quickFilter: 'quickFilter',
+    excelStyleFilter: 'excelStyleFilter'
+});
+export type FilterMode = (typeof FilterMode)[keyof typeof FilterMode];
 
-export enum GridSummaryPosition {
-    top = 'top',
-    bottom = 'bottom'
-}
+export const GridSummaryPosition = mkenum({
+    top: 'top',
+    bottom: 'bottom'
+});
+export type GridSummaryPosition = (typeof GridSummaryPosition)[keyof typeof GridSummaryPosition];
 
-export enum GridSummaryCalculationMode {
-    rootLevelOnly = 'rootLevelOnly',
-    childLevelsOnly = 'childLevelsOnly',
-    rootAndChildLevels = 'rootAndChildLevels'
-}
+export const GridSummaryCalculationMode = mkenum({
+    rootLevelOnly: 'rootLevelOnly',
+    childLevelsOnly: 'childLevelsOnly',
+    rootAndChildLevels: 'rootAndChildLevels'
+});
+export type GridSummaryCalculationMode = (typeof GridSummaryCalculationMode)[keyof typeof GridSummaryCalculationMode];
 
 export enum GridKeydownTargetType {
     dataCell = 'dataCell',
@@ -22,16 +26,18 @@ export enum GridKeydownTargetType {
     hierarchicalRow = 'hierarchicalRow'
 }
 
-export enum GridSelectionMode {
-    none = 'none',
-    single = 'single',
-    multiple = 'multiple',
-}
+export const GridSelectionMode = mkenum({
+    none: 'none',
+    single: 'single',
+    multiple: 'multiple'
+});
+export type GridSelectionMode = (typeof GridSelectionMode)[keyof typeof GridSelectionMode];
 
-export enum ColumnDisplayOrder {
-    Alphabetical = 'Alphabetical',
-    DisplayOrder = 'DisplayOrder'
-}
+export const ColumnDisplayOrder = mkenum({
+    Alphabetical: 'Alphabetical',
+    DisplayOrder: 'DisplayOrder'
+});
+export type ColumnDisplayOrder = (typeof ColumnDisplayOrder)[keyof typeof ColumnDisplayOrder];
 
 export enum ColumnPinningPosition {
     Start,

--- a/projects/igniteui-angular/src/lib/grids/grid-base.directive.ts
+++ b/projects/igniteui-angular/src/lib/grids/grid-base.directive.ts
@@ -964,7 +964,7 @@ export class IgxGridBaseDirective extends DisplayDensityBase implements
         return this._filterMode;
     }
 
-    set filterMode(value) {
+    set filterMode(value: FilterMode) {
         this._filterMode = value;
 
         if (this.filteringService.isFilterRowVisible) {
@@ -987,7 +987,7 @@ export class IgxGridBaseDirective extends DisplayDensityBase implements
         return this._summaryPosition;
     }
 
-    set summaryPosition(value) {
+    set summaryPosition(value: GridSummaryPosition) {
         this._summaryPosition = value;
         this.notifyChanges();
     }
@@ -1006,7 +1006,7 @@ export class IgxGridBaseDirective extends DisplayDensityBase implements
         return this._summaryCalculationMode;
     }
 
-    set summaryCalculationMode(value) {
+    set summaryCalculationMode(value: GridSummaryCalculationMode) {
         this._summaryCalculationMode = value;
         if (!this._init) {
             this.endEdit(true);
@@ -2579,7 +2579,7 @@ export class IgxGridBaseDirective extends DisplayDensityBase implements
     protected _hasVisibleColumns;
     protected _allowFiltering = false;
     protected _allowAdvancedFiltering = false;
-    protected _filterMode = FilterMode.quickFilter;
+    protected _filterMode: FilterMode = FilterMode.quickFilter;
 
     protected observer: ResizeObserver = new ResizeObserver(() => { });
 
@@ -2614,11 +2614,11 @@ export class IgxGridBaseDirective extends DisplayDensityBase implements
 
     protected _defaultTargetRecordNumber = 10;
 
-    private _summaryPosition = GridSummaryPosition.bottom;
-    private _summaryCalculationMode = GridSummaryCalculationMode.rootAndChildLevels;
-    private _cellSelectionMode = GridSelectionMode.multiple;
-    private _rowSelectionMode = GridSelectionMode.none;
-    private _columnSelectionMode = GridSelectionMode.none;
+    private _summaryPosition: GridSummaryPosition = GridSummaryPosition.bottom;
+    private _summaryCalculationMode: GridSummaryCalculationMode = GridSummaryCalculationMode.rootAndChildLevels;
+    private _cellSelectionMode: GridSelectionMode = GridSelectionMode.multiple;
+    private _rowSelectionMode: GridSelectionMode = GridSelectionMode.none;
+    private _columnSelectionMode: GridSelectionMode = GridSelectionMode.none;
 
     private rowEditPositioningStrategy = new RowEditPositionStrategy({
         horizontalDirection: HorizontalAlignment.Right,

--- a/projects/igniteui-angular/src/lib/progressbar/progressbar.component.ts
+++ b/projects/igniteui-angular/src/lib/progressbar/progressbar.component.ts
@@ -17,24 +17,26 @@ import {
     IgxProcessBarTextTemplateDirective,
     IgxProgressBarGradientDirective,
 } from './progressbar.common';
-import { IBaseEventArgs } from '../core/utils';
+import { IBaseEventArgs, mkenum } from '../core/utils';
 import { IgxDirectionality } from '../services/direction/directionality';
 
 const ONE_PERCENT = 0.01;
 const MIN_VALUE = 0;
 
-export enum IgxTextAlign {
-    START = 'start',
-    CENTER = 'center',
-    END = 'end'
-}
+export const IgxTextAlign = mkenum({
+    START: 'start',
+    CENTER: 'center',
+    END: 'end'
+});
+export type IgxTextAlign = (typeof IgxTextAlign)[keyof typeof IgxTextAlign];
 
-export enum IgxProgressType {
-    ERROR = 'error',
-    INFO = 'info',
-    WARNING = 'warning',
-    SUCCESS = 'success'
-}
+export const IgxProgressType = mkenum({
+    ERROR: 'error',
+    INFO: 'info',
+    WARNING: 'warning',
+    SUCCESS: 'success'
+});
+export type IgxProgressType = (typeof IgxProgressType)[keyof typeof IgxProgressType];
 
 export interface IChangeProgressEventArgs extends IBaseEventArgs {
     previousValue: number;

--- a/projects/igniteui-angular/src/lib/radio/radio.component.ts
+++ b/projects/igniteui-angular/src/lib/radio/radio.component.ts
@@ -8,7 +8,7 @@ import {
     ElementRef
 } from '@angular/core';
 import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
-import { isIE, IBaseEventArgs } from '../core/utils';
+import { isIE, IBaseEventArgs, mkenum } from '../core/utils';
 import { EditorProvider } from '../core/edit-provider';
 
 export interface IChangeRadioEventArgs extends IBaseEventArgs {
@@ -16,10 +16,12 @@ export interface IChangeRadioEventArgs extends IBaseEventArgs {
     radio: IgxRadioComponent;
 }
 
-export enum RadioLabelPosition {
-    BEFORE = 'before',
-    AFTER = 'after'
-}
+
+export const RadioLabelPosition = mkenum({
+    BEFORE: 'before',
+    AFTER: 'after'
+});
+export type RadioLabelPosition = (typeof RadioLabelPosition)[keyof typeof RadioLabelPosition];
 
 let nextId = 0;
 const noop = () => { };

--- a/projects/igniteui-angular/src/lib/switch/switch.component.ts
+++ b/projects/igniteui-angular/src/lib/switch/switch.component.ts
@@ -13,13 +13,14 @@ import {
 } from '@angular/core';
 import { CheckboxRequiredValidator, ControlValueAccessor, NG_VALIDATORS, NG_VALUE_ACCESSOR } from '@angular/forms';
 import { IgxRippleModule } from '../directives/ripple/ripple.directive';
-import { isIE, IBaseEventArgs } from '../core/utils';
+import { isIE, IBaseEventArgs, mkenum } from '../core/utils';
 import { EditorProvider } from '../core/edit-provider';
 
-export enum SwitchLabelPosition {
-    BEFORE = 'before',
-    AFTER = 'after'
-}
+export const SwitchLabelPosition = mkenum({
+    BEFORE: 'before',
+    AFTER: 'after'
+});
+export type SwitchLabelPosition = (typeof SwitchLabelPosition)[keyof typeof SwitchLabelPosition];
 
 export interface IChangeSwitchEventArgs extends IBaseEventArgs {
     checked: boolean;

--- a/projects/igniteui-angular/src/lib/tabs/tabs.component.ts
+++ b/projects/igniteui-angular/src/lib/tabs/tabs.component.ts
@@ -25,12 +25,13 @@ import { IgxTabsGroupComponent } from './tabs-group.component';
 import { IgxLeftButtonStyleDirective, IgxRightButtonStyleDirective, IgxTabItemTemplateDirective } from './tabs.directives';
 import { IgxTabsBase, IgxTabItemBase } from './tabs.common';
 import ResizeObserver from 'resize-observer-polyfill';
-import { PlatformUtil } from '../core/utils';
+import { mkenum, PlatformUtil } from '../core/utils';
 
-export enum IgxTabsType {
-    FIXED = 'fixed',
-    CONTENTFIT = 'contentfit'
-}
+export const IgxTabsType = mkenum({
+    FIXED: 'fixed',
+    CONTENTFIT: 'contentfit'
+});
+export type IgxTabsType = (typeof IgxTabsType)[keyof typeof IgxTabsType];
 
 let NEXT_TABS_ID = 0;
 

--- a/src/app/grid-groupby/grid-groupby.sample.ts
+++ b/src/app/grid-groupby/grid-groupby.sample.ts
@@ -75,7 +75,7 @@ export class GridGroupBySampleComponent implements OnInit {
             { 'Salary': 4900, 'ID': 'FRANS', 'CompanyName': 'Franchi S.p.A.', 'ContactName': 'Paolo Accorti', 'ContactTitle': 'Sales Representative', 'Address': 'Via Monte Bianco 34', 'City': 'Torino', 'Region': null, 'PostalCode': '10100', 'Country': 'Italy', 'Phone': '011-4988260', 'Fax': '011-4988261' }
         ];
     }
-    private _density = DisplayDensity.cosy;
+    private _density: DisplayDensity = DisplayDensity.cosy;
     public get density(): DisplayDensity {
         return this._density;
     }

